app/views/user_sessions/_new.html.erb

```diff
@@ -1,6 +1,6 @@
 <div class="container">
-
-<%= form_for :user_session, :as => :user_session, :url => :user_sessions, :html => {:class => "form col-md-6 offset-3"} do |f| %>
+<%= user_sessions_url + "?return_to=" + request.fullpath %>
```

Comment:
Please create a separate PR, for this so that we can also give you reward for this redirecting back changes.

Reply:
Also, open up an issue for it.